### ISSUE TITLE
Texture corruption in `sf::RenderTexture` when using `sf::View` Scissor test

### ISSUE THREAD

**Issue by SirusDoma** [SFML/SFML#3220]
### Prerequisite Checklist

- [X] I searched for [existing issues](https://github.com/search?q=repo%3ASFML%2FSFML&type=issues) to prevent duplicates
- [X] I searched for [existing discussions on the forum](https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fen.sfml-dev.org) to prevent duplicates
- [X] I am here to report an issue and not to just ask a question or look for help (use the [forum](https://en.sfml-dev.org/forums/index.php#c3) or [Discord](https://discord.gg/nr4X7Fh) instead)

### Describe your issue here

Due to smoothing issue in my game, I had to render all my drawables into `sf::RenderTexture` before finally render the texture into the window. This is temporary solution due to my assets problem, but this led me to discover strange behavior of the combination of `sf::View::setScissor` and `sf::RenderTexture`.

When using [scissor test](https://github.com/SFML/SFML/pull/1451) in render texture, the produced texture is somewhat corrupted, rendering the texture every frame led to flickering. The easy workaround to this problem is to draw something right after using the scissor view.

### Your Environment

- OS / distro / window manager: Windows
- SFML version: 5da286487ae79703c4b918c00f9542be67d8e660
- Compiler / toolchain: MSVC x86
- Special compiler / CMake flags: N/A

### Steps to reproduce

1. Initialize render texture
2. Use render texture with scissor view

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    auto window = sf::RenderWindow{sf::VideoMode{{200u, 200u}}, "Test"};
    auto renderTexture = sf::RenderTexture();

    if (!renderTexture.create(window.getSize()))
        return -1;

    auto shape = sf::CircleShape{100.f};
    shape.setFillColor(sf::Color::Green);

    auto left = window.getView();
    left.setScissor(sf::FloatRect{{0.f, 0.f}, {0.5f, 1.f}});

    auto right = window.getView();
    right.setScissor(sf::FloatRect{{0.5f, 0.f}, {0.5f, 1.f}});

    while (window.isOpen())
    {
        for (auto event = sf::Event{}; window.pollEvent(event);)
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }

        window.clear();
        {
            renderTexture.setView(window.getDefaultView());
            renderTexture.clear();

            renderTexture.setView(left);
            renderTexture.clear(sf::Color::Red);

            renderTexture.setView(right);
            renderTexture.draw(shape);

            // Uncomment this to workaround the issue
            // renderTexture.setView(renderTexture.getDefaultView()); // The bug will still occur if a different view is not applied
            // constexpr auto temp = std::array<sf::Vertex, 1>({sf::Vertex{{}, sf::Color::Transparent, {}}});
            // renderTexture.draw(&temp[0], temp.size(), sf::PrimitiveType::Points);

            renderTexture.display();
        }
        auto sprite = sf::Sprite(renderTexture.getTexture());
        window.draw(sprite);
        window.display();
    }

    return 0;
}
```
Note: The code is a slight modification from this PR: https://github.com/SFML/SFML/pull/1451

https://github.com/user-attachments/assets/4cfa5124-d214-4252-b9d2-982934e8f4ed

### Expected behavior

The produced texture from `sf::RenderTexture` should not be corrupted

### Actual behavior

The produced texture from `sf::RenderTexture` is corrupted

**Comment by eXpl0it3r** [comment]
This was most likely already fixed with: #3114

Can you try the latest master?

**Comment by SirusDoma** [comment]
Sorry, using the latest commit (7fdce1235e509242bfb9b3532be76830485e2a5e) fixes the problem.
Here the code I use to verify:

```cpp
#include <SFML/Graphics.hpp>

int main()
{
    auto window = sf::RenderWindow{sf::VideoMode{{200u, 200u}}, "Test"};
    auto renderTexture = sf::RenderTexture(window.getSize());

    auto shape = sf::CircleShape{100.f};
    shape.setFillColor(sf::Color::Green);

    auto left = window.getView();
    left.setScissor(sf::FloatRect{{0.f, 0.f}, {0.5f, 1.f}});

    auto right = window.getView();
    right.setScissor(sf::FloatRect{{0.5f, 0.f}, {0.5f, 1.f}});

    while (window.isOpen())
    {
        while (const auto event = window.pollEvent())
        {
            if (event->is<sf::Event::Closed>())
                window.close();
        }

        window.clear();
        {
            renderTexture.setView(window.getDefaultView());
            renderTexture.clear();

            renderTexture.setView(left);
            renderTexture.clear(sf::Color::Red);

            renderTexture.setView(right);
            renderTexture.draw(shape);

            renderTexture.display();
        }
        auto sprite = sf::Sprite(renderTexture.getTexture());
        window.draw(sprite);
        window.display();
    }

    return 0;
}
```
